src/main/java/teammates/common/datatransfer/StudentAttributesFactory.java

```diff
@@ -105,6 +105,13 @@ public StudentAttributes makeStudent(String enrollLine, String courseId) throws
 
         String paramSection;
 
+        /* Since the migration to spreadsheet interfaces, section header index will always be present.
+        Therefore, if the user chooses not to include a section entry in the student row during enrollment,
+        we treat it that the student has no section. */
+        if (columns[sectionColumnIndex].isEmpty()) {
+            hasSection = false;
+        }
+
```

Comment:
I don't like this approach, as I think these `has...` variables should purely reflect the status of the header row. You can just add a `&& !columns[sectionColumnIndex].isEmpty()` to the check below.